dataplane/saiserver/hostif.go

```diff
@@ -57,7 +59,19 @@ type hostif struct {
 	trapIDToHostifID map[uint64]uint64
 	groupIDToQueue   map[uint64]uint32
 	opts             *dplaneopts.Options
-	sendPortReq      func(msg *pktiopb.HostPortControlMessage) error
+	remoteMu         sync.Mutex
+	remoteHostifs    map[uint64]*pktiopb.HostPortControlMessage
+	remoteClosers    []func()
+	remotePortReq    func(msg *pktiopb.HostPortControlMessage) error
+}
+
+func (hostif *hostif) Reset() {
+	hostif.trapIDToHostifID = map[uint64]uint64{}
+	hostif.groupIDToQueue = map[uint64]uint32{}
+	hostif.remoteHostifs = map[uint64]*pktiopb.HostPortControlMessage{}
+	for _, closeFn := range hostif.remoteClosers {
```

Comment:
Do we need to clear remoteClosers too, and then re-init?

Reply:
done